### PR TITLE
Fixing #645

### PR DESCRIPTION
**Pull request by marstr**
This bypasses the Go Standard Library munging of Header casing.

### CODE REVIEW

**Comment by marstr** [comment]
Ah ha, so this is is interesting. CI is failing, even though test will pass against the live services. @mcardosos and I chased this down to being down to the fact that the recordings get tested in a case-sensitive way.

**Comment by marstr** [comment]
(For the sake of GitHub tagging, I'll repeat this information here.) This PR addresses #645

File: storage/client_test.go [hunk 1]
```diff
@@ -114,39 +138,55 @@ func compareHeaders(r *http.Request, i cassette.Request) bool {
...
 		}
-		requestHeaders.Del("Content-Type")
-		cassetteHeaders.Del("Content-Type")
+
+		deleteHeaders(requestHeaders, isContentType)
+		deleteHeaders(cassetteHeaders, isContentType)
 	}
 
-	return reflect.DeepEqual(requestHeaders, cassetteHeaders)
+	b := reflect.DeepEqual(requestHeaders, cassetteHeaders)
```
Comment (mcardosos) [inline storage/client_test.go]: >b := reflect.DeepEqual(requestHeaders, cassetteHeaders) [](start = 1, length = 55)

Better `return reflect.DeepEqual(requestHeaders, cassetteHeaders)`

Comment (marstr) [inline storage/client_test.go]: Derp, looks like debugging nonsense got included.

File: storage/client.go [hunk 2]
```diff
@@ -399,7 +399,7 @@ func (c Client) exec(verb, url string, headers map[string]string, body io.Reader
 	}
 
 	for k, v := range headers {
-		req.Header.Add(k, v)
+		req.Header[k] = append(req.Header[k], v)
```
Comment (jhendrixMSFT) [inline storage/client.go]: Can you add a comment here explaining why we don't use Add() here.

Comment (marstr) [inline storage/client.go]: Good call, we don't want history to repeat itself.

**Pull request merged**